test/ContextDependencyTemplateAsId.test.js

```diff
@@ -0,0 +1,140 @@
+"use strict";
+
+const _ = require("lodash");
+const should = require("should");
+const sinon = require("sinon");
+const ContextDependencyTemplateAsId = require("../lib/dependencies/ContextDependencyTemplateAsId");
+
+const requestShortenerMock = {
+	shorten: (request) => `shortened ${request}`
+};
+
+describe("ContextDependencyTemplateAsId", () => {
+	let env;
+
+	const applyContextDependencyTemplateAsId = function() {
+		const contextDependencyTemplateAsId = new ContextDependencyTemplateAsId();
+		const args = [].slice.call(arguments).concat(requestShortenerMock);
+		contextDependencyTemplateAsId.apply.apply(contextDependencyTemplateAsId, args);
+	};
+
+	beforeEach(() => {
+		env = {
+			source: {
+				replace: sinon.stub()
+			},
+			outputOptions: {
+				pathinfo: true
+			},
+			module: {
+				id: "123",
+				dependencies: [
+					"myModuleDependency"
+				]
+			},
+			baseDependency: {
+				range: [1, 25],
+				request: "myModule"
+			}
+		};
+	});
+
+	it("has apply function", () => {
+		(new ContextDependencyTemplateAsId()).apply.should.be.a.Function();
+	});
+
+	describe("when applied", () => {
+		describe("with module missing depedencies", () => {
+			beforeEach(() => {
+				applyContextDependencyTemplateAsId(env.baseDependency, env.source, env.outputOptions);
+			});
+
+			it("replaces source with missing module error", () => {
+				env.source.replace.callCount.should.be.exactly(1);
+				sinon.assert.calledWith(env.source.replace, 1, 24, '!(function webpackMissingModule() { var e = new Error("Cannot find module \\"myModule\\""); e.code = \'MODULE_NOT_FOUND\';; throw e; }())');
```

Comment:
@sokra I'm not sure I understand the problem with the test - this is the output from `WebpackMissingModule.js`
https://github.com/webpack/webpack/blob/a76ea83e3a47d4aece92bcc1979085d4756a5358/lib/dependencies/WebpackMissingModule.js#L15-L18

When boiled down, this will give:
```js
const errorCode = "var e = new Error('Cannot find module \"foo\"'}); e.code = 'MODULE_NOT_FOUND';";
return `${errorCode}; throw e;`;
```

The double `;;` comes from the output of `errorCode` having a trailing `;`, and the string it's interpolated in also having a trailing `;`